client/scripts/controllers/server/threads.ts

```diff
@@ -224,6 +249,83 @@ class ThreadsController {
     });
   }
 
+  public async fetchThread(id) {
+    const params = {
+      chain: app.activeChainId(),
+      community: app.activeCommunityId(),
+      id,
+    };
+    const response = await $.get(`${app.serverUrl()}/getThread`, params);
+    if (response.status !== 'Success') {
+      throw new Error(`Unsuccessful refresh status: ${response.status}`);
+    }
+    const thread = modelFromServer(response.result);
+    const existing = this._store.getByIdentifier(thread.id);
+    if (existing) this._store.remove(existing);
+    this._store.add(thread);
+    return modelFromServer(response.result);
+  }
+
+  // loadNextPage returns false if there are no more threads to load
+  public async loadNextPage(options: {
+    chainId: string,
+    communityId: string,
+    cutoffDate: moment.Moment,
+    topicId?: OffchainTopic
+  }) {
+    const { chainId, communityId, cutoffDate, topicId } = options;
+    const params = {
+      chain: chainId,
+      community: communityId,
+      cutoff_date: cutoffDate.toISOString(),
+    };
+    if (topicId) params['topic_id'] = topicId;
+    const response = await $.get(`${app.serverUrl()}/bulkThreads`, params);
+    if (response.status !== 'Success') {
+      throw new Error(`Unsuccessful refresh status: ${response.status}`);
+    }
+    const threads = (app.chain)
+      ? response.result.threads.filter((thread) => !thread.community)
+      : response.result.threads;
+    const comments = (app.chain)
+      ? response.result.comments.filter((comment) => !comment.community)
+      : response.result.comments;
+    for (const thread of threads) {
+      const modeledThread = modelFromServer(thread);
+      if (!thread.Address) {
+        console.error('OffchainThread missing address');
+      }
+      try {
+        if (topicId) {
+          if (this._topicListingStore.getById(thread.id)) {
+            this._topicListingStore.remove(this._topicListingStore.getById(thread.id));
+          }
+          this._topicListingStore.add(modeledThread);
+        } else {
+          if (this._listingStore.getByIdentifier(thread.id)) {
+            this._listingStore.remove(this._listingStore.getByIdentifier(thread.id));
+          }
+          this._listingStore.add(modeledThread);
+        }
+        this._store.add(modeledThread);
+      } catch (e) {
+        console.error(e.message);
+      }
+    }
+    for (const comment of comments) {
+      const existing = app.comments.store.getById(comment.id);
+      if (existing) {
+        app.comments.store.remove(existing);
+      }
+      try {
+        app.comments.store.add(modelCommentFromServer(comment));
+      } catch (e) {
+        console.error(e.message);
+      }
+    }
+    return !(threads.length < 20);
```

Comment:
why is there a magic number here? it seems like this has some interdependency with the backend - if so, we should make that very clear here and in the corresponding backend code

Reply:
This "magic number" is the size of a page of posts, the same size that the SQL query maxes out at. So if the fetch returns less than that query cutoff, it means we're out of relevant posts. This allows functions calling loadNextPage to know when to stop calling it ASAP. I'll add a comment for clarity.

Reply:
i think we should pull it out into an appropriate named constant at the top of the call - this is definitely one place where we could have unexpected behavior if someone changes the parameter at one end but not the other. we should also add a comment in the SQL query/queries where the page size is defined (and probably factor that out into a const too)

Reply:
What about if the bulkThreads call took a page size, passed in via the loadNextPage fn, such that the two values (the fn return and the sql limit) were necessarily coupled?

Reply:
that works too! it does need a reasonable max though